### ISSUE TITLE
ESLint can't detect error from typescript

### ISSUE THREAD

**Issue by zarkin404** [typescript-eslint/typescript-eslint#352]
**Repro**

```Javascript
module.exports = {
  root: true,
  env: {
    browser: true
  },
  parserOptions: {
    parser: '@typescript-eslint/parser'
  },
  extends: ['plugin:vue/strongly-recommended', 'plugin:prettier/recommended'],
  plugins: ['@typescript-eslint', 'vue', 'prettier'],
  rules: {
    'prettier/prettier': ['error', { semi: false, singleQuote: true }],
    '@typescript-eslint/class-name-casing': 'error',
    'vue/max-attributes-per-line': 'off',
    'vue/html-self-closing': 'off'
  }
}
```

```TS
    let a = {}
    console.log(a.b)
```

**Expected Result**
While running build task, an error can be raised to notice me and stop the compilation.

**Actual Result**
building run successfully and no error raised.

**Additional Info**
VSCode Problems Panel can show that `Property 'b' does not exist on type '{}'.` But when i run `eslint --ext .js,.ts,.vue --ignore-path .gitignore .` , i saw that eslint pass the checking with `Done in 3.73s.`

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.4.2` |
| `@typescript-eslint/parser`        | `1.4.2` |
| `eslint-config-prettier`        | `4.1.0` |
| `eslint-plugin-prettier`        | `3.0.1` |
| `eslint-plugin-vue`        | `5.2.2` |
| `prettier`                        | `1.16.4` |
| `ts-loader`                        | `5.3.3` |
| `npm`                              | `6.4.1` |
| `TypeScript`                       | `3.2.4` |
| `ESLint`                           | `5.15.0` |
| `node`                             | `10.14.1` |
| `npm`                              | `6.4.1` |

**Comment by mohsen1** [comment]
It's by design I believe. ESLint is not in business of type checking your code. You should rely on TypeScript for checking for type errors.

**Comment by bradzacher** [comment]
@mohsen1 is correct, we purposely do not care about errors that should arise as a result of type checking errors.
We only care that the code is *syntactically* valid (i.e. can be parsed without errors), not about whether it's *semantically* valid (i.e. don't care if it is type sound).

This plugin handles linting of your code only - meaning we care about parsing the code, and ensuring it adheres to a set of opinionated patterns related to style and structure.

If you want your code to be typechecked - please use `tsc` (or one of the tools built for your specific build chain).

**Comment by zarkin404** [comment]
It seems that i misunderstand the usage of ESLint.:sweat_smile: BTW, thanks for reminding me of checking my build config, i found that the option `[transpileOnly](https://github.com/TypeStrong/ts-loader#loader-options)` of the `ts-loader` will make project lost static type checking. After switching it to false, the build task can be interupt successfully.

**Comment by bradzacher** [comment]
A suggestion for using ts-loader is to also use https://github.com/Realytics/fork-ts-checker-webpack-plugin

I have found in the past that this speeds up up your build because it parallelises the type checking and compilation steps.

**Comment by viT-1** [comment]
@bradzacher Strange cycle link =) From typescript-eslint to webpack plugin (I don't use webpack) to [tslint deprecation](https://github.com/TypeStrong/fork-ts-checker-webpack-plugin#tslint)... and viola: we returned to eslint again! ;)

**Comment by viT-1** [comment]
I have found only one workaround: use tsc as linter
in package.json scripts:
`"lint": "tsc --project ./src/tsconfig.dev.json --noEmit && eslint **/*.{ts,json,js} *.*"`
